src/dust/dust.cpp

```diff
@@ -168,8 +188,38 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   }
 
   // Dust density
-  params.Add("grain_density", pin->GetOrAddReal("dust", "grain_density", 1.0));
+  const Real rho_p_orig = pin->GetOrAddReal("dust", "grain_density", 1.0);
+  params.Add("grain_density", rho_p_orig);
 
+  if (enable_dust_coagulation) {
+    if (cgsunit == NULL) {
+      cgsunit = new CGSUnit;
+    }
+    if (!cgsunit->isSet()) {
+      cgsunit->SetCGSUnit(pin);
+    }
+    params.Add("cgs_unit", cgsunit);
+
+    Real rho_p = rho_p_orig;
+    // density in physical unit correspinding to code unit 1.
+    // const Real rho_g0 = pin->GetOrAddReal("dust", "rho_g0", 1.0);
+    const Real rho_g0 = cgsunit->mass0 / cgsunit->vol0;
+    // re-scale rho_p to account for rho_g0 for stopping_time calculation
+    rho_p /= rho_g0;
+
+    // re-scale rho_p by prefactor coefficient
+    if (cgsunit->isurface_den) {
+      rho_p *= 0.5 * M_PI;
+    } else {
+      rho_p *= std::sqrt(M_PI / 8.);
+      rho_p /= cgsunit->length0;
+    }
+
+    if (parthenon::Globals::my_rank == 0) {
+      std::cout << "rho_p, rho_g0=" << rho_p << " " << rho_g0 << std::endl;
+    }
+    params.Add("rho_p", rho_p);
```

Comment:
See earlier comment about not doing unit conversions

Reply:
fixed in the new committed code